File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_py/code_output/MusicPlayer.py
import random


class MusicPlayer:
    def __init__(self):
        self.playlist = []
        self.currentSong = None
        self.volume = 50

    def addSong(self, song):
        self.playlist.append(song)

    def removeSong(self, song):
        if song in self.playlist:
            self.playlist.remove(song)
            if self.currentSong is not None and self.currentSong == song:
                self.stop()

    def play(self):
        if not self.playlist:
            return None
        if self.currentSong is not None:
            for song in self.playlist:
                if song == self.currentSong:
                    return self.currentSong
        return self.playlist[0]

    def stop(self):
        if self.currentSong is not None:
            self.currentSong = None
            return True
        else:
            return False

    def switchSong(self):
        if self.currentSong is None:
            return False
        currentIndex = self.playlist.index(self.currentSong)
        if currentIndex < len(self.playlist) - 1:
            self.currentSong = self.playlist[currentIndex + 1]
            return True
        else:
            return False

    def previousSong(self):
        if self.currentSong is None:
            return False
        currentIndex = self.playlist.index(self.currentSong)
        if currentIndex > 0:
            self.currentSong = self.playlist[currentIndex - 1]
            return True
        else:
            return False

    def setVolume(self, volume):
        if 0 <= volume <= 100:
            self.volume = volume
            return True
        else:
            return False

    def shuffle(self):
        if not self.playlist:
            return False
        random.shuffle(self.playlist)
        return True

    def getPlaylist(self):
        return self.playlist

    def setPlaylist(self, playlist):
        self.playlist = playlist

    def getCurrentSong(self):
        return self.currentSong

    def setCurrentSong(self, currentSong):
        self.currentSong = currentSong

    def getVolume(self):
        return self.volume

File: data/derived_cot/rq1_traces/translation/r1/java_to_cpp/code_output/MusicPlayer.cpp
#include <vector>
#include <string>
#include <algorithm>
#include <random>

class MusicPlayer {
private:
    std::vector<std::string> playlist;
    std::string currentSong;
    int volume;

public:
    MusicPlayer() : currentSong(""), volume(50) {}

    void addSong(const std::string& song) {
        playlist.push_back(song);
    }

    void removeSong(const std::string& song) {
        auto it = std::find(playlist.begin(), playlist.end(), song);
        if (it != playlist.end()) {
            playlist.erase(it);
            if (currentSong == song) {
                stop();
            }
        }
    }

    std::string play() {
        if (playlist.empty()) {
            return "";
        }
        if (!currentSong.empty()) {
            for (const std::string& s : playlist) {
                if (s == currentSong) {
                    return currentSong;
                }
            }
        }
        return playlist[0];
    }

    bool stop() {
        if (!currentSong.empty()) {
            currentSong = "";
            return true;
        } else {
            return false;
        }
    }

    bool switchSong() {
        if (currentSong.empty()) {
            return false;
        }
        auto it = std::find(playlist.begin(), playlist.end(), currentSong);
        if (it == playlist.end()) {
            return false;
        }
        int currentIndex = it - playlist.begin();
        if (currentIndex < static_cast<int>(playlist.size()) - 1) {
            currentSong = playlist[currentIndex + 1];
            return true;
        } else {
            return false;
        }
    }

    bool previousSong() {
        if (currentSong.empty()) {
            return false;
        }
        auto it = std::find(playlist.begin(), playlist.end(), currentSong);
        if (it == playlist.end()) {
            return false;
        }
        int currentIndex = it - playlist.begin();
        if (currentIndex > 0) {
            currentSong = playlist[currentIndex - 1];
            return true;
        } else {
            return false;
        }
    }

    bool setVolume(int volume) {
        if (volume >= 0 && volume <= 100) {
            this->volume = volume;
            return true;
        } else {
            return false;
        }
    }

    bool shuffle() {
        if (playlist.empty()) {
            return false;
        }
        std::random_device rd;
        std::mt19937 g(rd());
        std::shuffle(playlist.begin(), playlist.end(), g);
        return true;
    }

    std::vector<std::string>& getPlaylist() {
        return playlist;
    }

    void setPlaylist(const std::vector<std::string>& playlist) {
        this->playlist = playlist;
    }

    std::string getCurrentSong() const {
        return currentSong;
    }

    void setCurrentSong(const std::string& currentSong) {
        this->currentSong = currentSong;
    }

    int getVolume() const {
        return volume;
    }
};